lightning/src/ln/outbound_payment.rs

```diff
@@ -1048,6 +1049,11 @@ impl OutboundPayments {
 						keysend_preimage,
 						retry_strategy: *retry_strategy,
 						route_params,
+						invoice_request:
+							retryable_invoice_request
+							.take()
+							.ok_or(Bolt12PaymentError::UnexpectedInvoice)?
+							.invoice_request,
```

Comment:
This may be `None` if the `InvoiceRequest` was already retried, so we'll need to have some other means of tracking whether or not to retry. I think this was mentioned in an earlier PR, so perhaps it will be handled in a follow-up?

Reply:
Ah yeah, probably will do that in a follow-up. Looks like it's tracked here https://github.com/lightningdevkit/rust-lightning/issues/2298. Thanks for the reminder!